src/Reveal.Sdk.Dom.Tests/Visualizations/VisualizationFixture.cs

```diff
@@ -0,0 +1,445 @@
+
+            // Act
+            var mock = new Mock<Visualization>("testTitle", dataSourceItem) { CallBase = true };
+            var visualization = mock.Object;
+
+            // Assert
+            Assert.IsType<TabularDataDefinition>(visualization.DataDefinition);
+            Assert.Equal(dataSourceItem, visualization.DataDefinition.DataSourceItem);
+            Assert.Equal(dataSourceItem.Fields.Count, (visualization.DataDefinition as TabularDataDefinition).Fields.Count);
+            Assert.Equal(dataSourceItem.JoinTables.Count, (visualization.DataDefinition as TabularDataDefinition).JoinTables.Count);
+        }
+
+        [Fact]
+        public void Constructor_UseXmlaDataDefinition_WhenNotUseTabularDataSource()
+        {
+            // Arrange
+            var dataSource = new DataSource();
+            var dataSourceItem = new DataSourceItem
+            {
+                DataSource = dataSource,
+                HasTabularData = false,
+            };
+
+            // Act
+            var visualization = new Mock<Visualization>("testTitle", dataSourceItem) { CallBase = true }.Object;
+
+            // Assert
+            Assert.IsType<XmlaDataDefinition>(visualization.DataDefinition);
+        }
+
+        [Fact]
+        public void Id_IsUnique_WhenConstructed()
+        {
+            // Arrange
+            var dataSource = new DataSource();
+            var dataSourceItem = new DataSourceItem
+            {
+                DataSource = dataSource,
+                HasTabularData = false,
+            };
+
+            // Act
+            var visualization = new Mock<Visualization>("testTitle", dataSourceItem) { CallBase = true }.Object;
+            var visualization2 = new Mock<Visualization>("testTitle2", dataSourceItem) { CallBase = true }.Object;
+
+            // Assert
+            Assert.NotEmpty(visualization.Id);
+            Assert.NotEmpty(visualization2.Id);
+            Assert.NotEqual(visualization.Id, visualization2.Id);
+        }
+
+        [Fact]
+        public void Id_GetSetCorrectly_WhenUsed()
+        {
+            // Arrange
+            var dataSource = new DataSource();
+            var dataSourceItem = new DataSourceItem
+            {
+                DataSource = dataSource,
+                HasTabularData = false,
+            };
+            var visualization = new Mock<Visualization>("testTitle", dataSourceItem) { CallBase = true }.Object;
+
+
+            // Act
+            visualization.Id = "testId";
+
+            // Assert
+            Assert.Equal("testId", visualization.Id);
+        }
+
+        [Fact]
+        public void Title_GetSetCorrectly_WhenUsed()
+        {
+            // Arrange
+            var dataSource = new DataSource();
+            var dataSourceItem = new DataSourceItem
+            {
+                DataSource = dataSource,
+                HasTabularData = false,
```

Comment:
We should only set HasTabularData value here in case we want to write unit test for its specific value